src/services/VerifyAccountRequestProcessor.ts

```diff
@@ -210,14 +212,14 @@ export class VerifyAccountRequestProcessor {
   		return Response(HttpCodesEnum.UNAUTHORIZED, "Too many attempts");
   	}
 
-  	const name = getFullName(person.name);
+  	const name = getNameByType(person.name);
   	this.logger.appendKeys({ govuk_signin_journey_id: session.clientSessionId });
   	const timeOfRequest = absoluteTimeNow();
 
-  	let { hmrcUuid } = session;
-  	if (!hmrcUuid) {
-  		hmrcUuid = randomUUID();
-  		await this.BavService.saveHmrcUuid(sessionId, hmrcUuid);
+  	let { vendorUuid } = session;
```

Comment:
Is this a duplicate?

Reply:
Yes, we perform the same check + react accordingly to add a uuid for HMRC - I renamed to the generic vendorUuid so the session structure remains the same, no need for 2 uuid's (one for Experian and one for HMRC)

Reply:
Should this be a private method / function then to avoid duplicating the code

Reply:
Added function to cover the above